activerecord/lib/active_record/test_databases.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "active_support/testing/parallelization"
+
+module ActiveRecord
+  module TestDatabases
+    ActiveSupport::Testing::Parallelization.after_fork_hook do |i|
+      create_and_migrate(i, spec_name: Rails.env)
+    end
+
+    ActiveSupport::Testing::Parallelization.run_cleanup_hook do |i|
+      drop(i, spec_name: Rails.env)
+    end
+
+    def self.create_and_migrate(i, spec_name:)
+      old, ENV["VERBOSE"] = ENV["VERBOSE"], "false"
+
+      connection_spec = ActiveRecord::Base.configurations[spec_name]
+
+      connection_spec["database"] += "-#{i}"
+      ActiveRecord::Tasks::DatabaseTasks.create(connection_spec)
+      ActiveRecord::Base.establish_connection(connection_spec)
+      if ActiveRecord::Base.connection.migration_context.needs_migration?
+        ActiveRecord::Tasks::DatabaseTasks.migrate
+      end
+    ensure
+      ActiveRecord::Base.establish_connection(ActiveRecord::Base.configurations[Rails.env])
```

Comment:
It is just question: can't realize why do we need re-establish the connection?

Reply:
We need to re-establish the original connection to AR Base since we want the `test` connection, not the other db connection. Otherwise the tests will try to all run against AR Base with the other db connection, not the `test` connection.